astropy/io/fits/tests/test_hdulist.py

```diff
@@ -782,3 +782,31 @@ def test_extname_in_hdulist(self):
         assert ('a', 2) not in hdulist
         assert ('b', 1) not in hdulist
         assert ('b', 2) not in hdulist
+
+    def test_overwrite_vs_clobber(self):
+        hdulist = fits.HDUList([fits.PrimaryHDU()])
+        hdulist.writeto(self.temp('test_overwrite.fits'))
+        hdulist.writeto(self.temp('test_overwrite.fits'), overwrite=True)
+        with catch_warnings(AstropyDeprecationWarning) as warning_lines:
+            hdulist.writeto(self.temp('test_overwrite.fits'), clobber=True)
+            assert warning_lines[0].category == AstropyDeprecationWarning
+            assert (str(warning_lines[0].message) == "The use of the "
+                    "clobber parameter is deprecated. Please use "
+                    "overwrite in the future.")
+        # All four illegal combinations
```

Comment:
I don't think we should check these cases (those 4 illegal combinations). They are not part of the `io.fits` functions anymore and if (or as soon as) we remove clobber completly we will have to remove these anyway.